### ISSUE TITLE
how-to-guide: expanding the GUI withe viewer dock widgets

### ISSUE THREAD

**Issue by sofroniewn** [napari/tutorials#68]
Write a how to guide on customizing the viewer by adding viewer dock widgets - see post here https://forum.image.sc/t/integration-of-napari-module-subclass-plugin/36018/2?u=sofroniewn and discussion here https://github.com/napari/napari/pull/1122#issuecomment-611595540.

Maybe @tlambert03 can take lead

**Comment by tlambert03** [comment]
yeah, I've been meaning to do that, and also probably an overview on event handling similar to comments in https://github.com/napari/napari/issues/1110#issuecomment-609057214 and https://github.com/napari/napari/issues/1110#issuecomment-609078376

I'll also add https://github.com/napari/tutorials/issues/52#issuecomment-582094724, so I don't forget about it.

I'm still conflicted on whether this stuff belongs in tutorials or sphinx docs though (https://github.com/napari/napari/issues/1049).  I like the tutorials as a "hey what is napari anyway?"  but I'm conflicted on the format for things like this.

**Comment by sofroniewn** [comment]
> I'm still conflicted on whether this stuff belongs in tutorials or sphinx docs though (napari/napari#1049). I like the tutorials as a "hey what is napari anyway?" but I'm conflicted on the format for things like this.

This https://github.com/napari/napari/issues/764#issuecomment-605077118 sums up my current thinking on the different content we want. To me these "how-to-guides" are different from our "API reference" and our "tutorials". I guess as we go further with the website mocks it will maybe be more clear where what should live / whether it should be in sphinx or not.

I guess in terms of writing this stuff the difference is `.rst` vs `.md` and how you embed some of the content? If you can just write it as `.md` then I'd just to that. If you find yourself wanting the `.rst` and other features then that's a big sign.

I'd say for now more important to think about getting content in in parallel to the website re-organization and then when we have more content it will be easier to group and see where it should go long term. Does that help?

**Comment by tlambert03** [comment]
your last point is the most important one...  we can definitely reorganize later.

but just for the sake of argument... I guess I'm still not in love with those 4 distinctions.  they feel slightly arbitrary rather than "the obvious 4 things".  tutorials and how-to-guides are extremely similar.  And actually I feel like this particular topic sits more on the edge of how-to and explanations... both of which I think I'd prefer to put in sphinxdocs. 

the issue with rst vs md is mostly about how often we want to be linking to our own code.  I _much_ prefer writing md, but it doesn't let you do something like 
`:meth:<~napari.components.add_layers_mixin.AddLayersMixin.add_image>` and have it render nicely as [**`add_image()`**](https://napari.org/docs/api/napari.components.html#napari.components.add_layers_mixin.AddLayersMixin.add_image) with a direct link to the appropriate docstring in the API reference (regardless of where the rst file actually lives).

but this is peripheral... and I'll get on writing something up :)
